### PR TITLE
Fix: Active request fixes and refactor

### PR DESCRIPTION
**Pull request by Tulun**
**Motivation**
Breakout and cleanup ActiveRequests and ActiveRequestsForm code

**Summary**
Renamed and broke out request form into Commit and Reveal Phase. Snapshot button now renders in Reveal Phase and swaps between reveal votes.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [x]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: src/features/vote/SubmitModal.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  reset: (
+    values?:
+      | {
+          [x: string]: string | undefined;
+        }
+      | undefined,
+    omitResetState?:
+      | Partial<{
+          errors: boolean;
+          isDirty: boolean;
```
Comment (Gamaranto) [inline src/features/vote/SubmitModal.tsx:38]: What does `isDirty` does? It doesn't seem used anywhere

**Review by Gamaranto — APPROVED** [review]
just a minor nit, otherwise LGTM

File: src/features/vote/SubmitModal.tsx (hunk 1 above)
Comment (Tulun) [reply src/features/vote/SubmitModal.tsx:38]: Ripped from the lib's typescript. No idea.

File: src/features/vote/RevealPhase.tsx [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+                    }
+                  });
+
+                  // Make sure to use the two key contract for revealing if it exists
+                  let vc = votingContract;
+                  if (designatedVotingContract) vc = designatedVotingContract;
+                  if (vc) {
+                    revealVotes(vc, postData).then((res) => {
+                      // refetch votes.
+                      console.log("successfully revealed");
```
Comment (daywiss) [inline src/features/vote/RevealPhase.tsx:267]: can prob remove this

File: src/features/vote/RevealPhase.tsx [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+    }
+    if (activeRequests.length && encryptedVotes.length) {
+      const tv = [] as TableValue[];
+      activeRequests.forEach((el) => {
+        const datum = {} as TableValue;
+        datum.ancillaryData = el.ancillaryData;
+        datum.identifier = el.identifier;
+        let vote = "-";
+        // I believe latest events are on bottom. requires testing.
+        const latestVotesFirst = [...encryptedVotes].reverse();
```
Comment (daywiss) [inline src/features/vote/RevealPhase.tsx:105]: this should probably be pulled out of the loop

File: src/features/vote/RevealPhase.tsx [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+              type="button"
+              onClick={() => {
+                // WIP. Comment out for now.
+                // console.log("encryptedVotes", encryptedVotes);
+                if (encryptedVotes.length && activeRequests.length) {
+                  const postData = [] as PostRevealData[];
+                  activeRequests.forEach((el, index) => {
+                    const datum = {} as PostRevealData;
+                    // I believe latest events are on bottom. requires testing.
+                    const latestVotesFirst = [...encryptedVotes].reverse();
```
Comment (daywiss) [inline src/features/vote/RevealPhase.tsx:234]: probably want to move this out of the loop

File: src/features/vote/RevealPhase.tsx [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+                    const findVote = latestVotesFirst.find(
+                      (x) => x.identifier === el.identifier
```
Comment (daywiss) [inline src/features/vote/RevealPhase.tsx:236]: do we not need to match against more fields here?

File: src/features/vote/RevealPhase.tsx [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+        );
+        if (findReveal) {
+          datum.revealed = true;
+        } else {
+          datum.revealed = false;
+        }
+
+        tv.push(datum);
+      });
+      setTableValues(tv);
```
Comment (daywiss) [inline src/features/vote/RevealPhase.tsx:136]: can we also save revealableVotes here, and remove the logic from when the reveal button is pressed?

File: src/features/vote/SubmitModal.tsx [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+            )}
+          </div>
+        </ModalWrapper>
+      </Modal>
+    </>
+  );
+};
+
+const SubmitModal = forwardRef(_SubmitModal);
+SubmitModal.displayName = "SubmitModal";
```
Comment (daywiss) [inline src/features/vote/SubmitModal.tsx:154]: just a small nit with this file, `SubmitModal` is so generic, took me a while to realize it was doing a specific thing. would be great if we could make this more specific like `SubmitVotesModal` or `CommitVotesModal` or something like that.

**Review by daywiss — COMMENTED** [review]
look good, just have some minor nits, and maybe attempt to clean up the reveal phase logic

File: src/features/vote/RevealPhase.tsx (hunk 2 above)
Comment (Tulun) [reply src/features/vote/RevealPhase.tsx:267]: I actually do some stuff in here (refetch data) in my next branch. I'll just leave it alone.

File: src/features/vote/RevealPhase.tsx (hunk 5 above)
Comment (Tulun) [reply src/features/vote/RevealPhase.tsx:236]: Good call, missed this one.

File: src/features/vote/RevealPhase.tsx (hunk 6 above)
Comment (Tulun) [reply src/features/vote/RevealPhase.tsx:136]: Great idea.

File: src/features/vote/SubmitModal.tsx (hunk 7 above)
Comment (Tulun) [reply src/features/vote/SubmitModal.tsx]: Good call.

File: src/features/vote/RevealPhase.tsx [hunk 8]
```diff
@@ -0,0 +1,262 @@
...
+              type="button"
+              onClick={() => {
+                console.log("PSR", postRevealData);
+                // Make sure to use the two key contract for revealing if it exists
+                let vc = votingContract;
+                if (designatedVotingContract) vc = designatedVotingContract;
+                if (vc) {
+                  revealVotes(vc, postRevealData).then((res) => {
+                    // refetch votes.
+                    console.log("successfully revealed");
```
Comment (daywiss) [inline src/features/vote/RevealPhase.tsx]: i dont wnat to forget to remove these logs, line 232 and 214.  We might want a TODO on having a way to notify user of success

**Review by daywiss — APPROVED** [review]
looks really good, thanks for taking the time to refactor this. i think just remove logs for now, and we can push more extreme changes to a later point

**Pull request merged**